lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/time/Time.java

```diff
@@ -18,53 +18,61 @@
  */
 package org.apache.reef.wake.time;
 
+import java.util.Date;
+
 /**
- * Time object.
+ * An abstract object that has a timestamp.
+ * That allows up to compare and order objects by the time.
  */
 public abstract class Time implements Comparable<Time> {
 
   private final long timestamp;
 
+  /** Cached version of toString() result. */
+  private String timestampStr = null;
+
+  /**
+   * Initialize the internal timestamp. Timestamp remains constant
+   * for the entire lifecycle of the object.
+   * @param timestamp timestamp in milliseconds since the beginning
+   * of the epoch (01/01/1970).
+   */
   public Time(final long timestamp) {
     this.timestamp = timestamp;
   }
 
-  public final long getTimeStamp() {
+  /**
+   * Get timestamp in milliseconds since the beginning of the epoch (01/01/1970).
+   * @return Object's timestamp in milliseconds since the start of the epoch.
+   */
+  public final long getTimestamp() {
```

Comment:
Good point. I've marked the original method as deprecated and opened [REEF-1532](https://issues.apache.org/jira/browse/REEF-1532) in JIRA